### PR TITLE
Unsubscribe function and confirmation

### PR DESCRIPTION
**Pull request by debugwand**

### CODE REVIEW

File: functions/unsubscribe/lib/confirmation/auth-token.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+const logger = require('@financial-times/n-lambda').logger;
+const metrics = require('../../lib-shared/metrics');
+const tap = require('../../lib-shared/utils/tap');
+const cookie = require('cookie');
+const url = require('url');
+const AUTH_URL = `${process.env.FT_DEVELOPER_BETA_URL}/authorize?client_id=${process.env.FT_DEVELOPER_CLIENT_NAME}&response_type=token`;
+
+
+module.exports = function authToken (cookieString) {
+	const FTSessionCookie = cookie.parse(cookieString).FTSession_s;
```
Comment (phamann) [inline functions/unsubscribe/lib/confirmation/auth-token.js]: What happens if a user/request doesn't have this cookie?

File: functions/unsubscribe/lib/confirmation/get-user-email.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+module.exports = function getUserEmail (uuid, authToken) {
+	const USER_URL = `${process.env.FT_DEVELOPER_BETA_URL}/users/${uuid}/profile`;
+	const options = {
+		headers: {
+			'Authorization': `Bearer ${authToken}`,
+			'X-Api-Key': process.env.FT_DEVELOPER_API_KEY
+		},
+		redirect: 'manual',
+		timeout: 500
```
Comment (phamann) [inline functions/unsubscribe/lib/confirmation/get-user-email.js]: Is this intentionally low?

File: functions/unsubscribe/lib/confirmation/auth-token.js (hunk 1 above)
Comment (debugwand) [inline functions/unsubscribe/lib/confirmation/auth-token.js]: If they're not authorised then the entrypoint in index.js will call `auth.rejectWithError` before they even get this far.

File: functions/unsubscribe/lib/confirmation/get-user-email.js (hunk 2 above)
Comment (debugwand) [inline functions/unsubscribe/lib/confirmation/get-user-email.js]: Its the same as what was being used by another team, what would you normally consider sensible?

File: functions/unsubscribe/lib/confirmation/render.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+		],
+		layoutsDir: path.join(bowerDirectory, 'n-ui/layout'),
+		defaultLayout: 'vanilla',
+		directory: directory
+	})
+	.then(handlebars => {
+		const fullPath = path.join(directory, view);
+		// Currently renderView does not return a promise. See:
+		// https://github.com/ericf/express-handlebars/issues/185
+		return new Promise((resolve, reject) => {
```
Comment (phamann) [inline functions/unsubscribe/lib/confirmation/render.js]: our handlebars abstraction `n-handlebars` does expose a promise friendly api. See: https://github.com/Financial-Times/next-myft-email/blob/master/functions/create-email/lib/email/render.js#L21-L23

File: functions/unsubscribe/lib/confirmation/get-user-email.js (hunk 2 above)
Comment (phamann) [inline functions/unsubscribe/lib/confirmation/get-user-email.js]: Within next our default is `3000`, very lenient I know, but for the sake of email this is acceptable.

**Comment by phamann** [comment]
👍 LGTM other than comments

**Pull request merged**